compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingTypes.h

```diff
@@ -63,6 +63,29 @@ MatmulNarrowDim getMatmulNarrowDim(linalg::LinalgOp linalgOp,
                                    int narrowThreshold);
 
 // The structs defined here because they are used by encoding_interfaces.td.
+
+/// Bundles an encoding attribute with its associated dynamic information.
+/// This is returned by interface methods that query encoding properties
+/// from operations. Dynamic values can include runtime information needed
+/// by the encoding (e.g., M, N, K dimensions for matmul encodings).
+struct EncodingProperties {
+  /// The encoding attribute for the operand/result.
+  Attribute encoding;
+  /// Dynamic values needed by the encoding. For matmul-like operations,
+  /// these typically correspond to the iteration domain dimensions (M, N, K).
+  /// TODO(jtuyls): Currently empty; will be populated to support dynamic
+  /// layout decisions. See https://github.com/iree-org/iree/issues/22370.
+  SmallVector<Value> dynamicValues;
+};
+
+/// Bundles encoding properties for all operands and results of an operation.
+/// This is returned by SerializableAttr::getEncodingProperties() when
+/// deriving encodings from an operation (e.g., linalg.matmul).
+struct OpEncodingProperties {
+  SmallVector<EncodingProperties> operands;
+  SmallVector<EncodingProperties> results;
+};
```

Comment:
I'm thinking if we use `inputs` and `inits`, so it connects to DPS ops better. `results` looks a little wonky to me, because you don't set encodings on results. (We unset encodings on results, though.) What do you think?